src/constructs/__tests__/named-capture.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { buildRegExp, namedCapture, oneOrMore } from '../..';
+
+describe('namedCapture function', () => {
+  it('should create a named capture group', () => {
+    const regex = buildRegExp(namedCapture('a', 'group1'));
+    const match = regex.exec('a');
+    expect(match?.groups?.group1).toBe('a');
+  });
+
+  it('should not match when the named capture group does not match the input', () => {
+    const regex = buildRegExp(namedCapture('a', 'group1'));
+    const match = regex.exec('b');
+    expect(match).toBeNull();
+  });
+
+  it('should handle multiple named capture groups', () => {
+    const regex = buildRegExp(namedCapture(['a', namedCapture('b', 'group2')], 'group1'));
+    const match = regex.exec('ab');
+    expect(match).not.toBeNull();
+    expect(match?.groups?.group1).toBe('ab');
+    expect(match?.groups?.group2).toBe('b');
+  });
+
+  it('should handle nested named capture groups', () => {
+    const regex = buildRegExp(namedCapture(['a', namedCapture('b', 'group2')], 'group1'));
+    const match = regex.exec('ab');
+    expect(match?.groups?.group1).toBe('ab');
+    expect(match?.groups?.group2).toBe('b');
+  });
+});
+
+describe('namedCapture RegEx matching', () => {
+  test('`named-capture` pattern', () => {
+    expect(namedCapture('a', 'abba')).toEqualRegex(/(?<abba>a)/);
```

Comment:
When seeing `namedCapture('a', 'abba')` it's hard to tell which part is the matched pattern and which one is the name of capturing group.